### ISSUE TITLE
Database move from binary to source

### ISSUE THREAD

**Issue by CaptainBobby** [gophish/gophish#189]
I recently moved from a binary install to source install in order to have additional features. I wasn't sure about the process of getting the existing database in the source install so I just guessed and moved the gophish.db in to the source directory. Once connected everything seemed hunky dory! All my email templates, landing pages, target groups, dashboard data, everything was there. Then I clicked to view the details of a campaign and it says "Campaign not found!"

Does the campaign data exist somewhere else? Am I supposed to be migrating the data in some other way. Any help with this would be appreciated.

![campaigns](https://cloud.githubusercontent.com/assets/17417959/13504867/a2ee74a8-e133-11e5-90cd-73efdcff5e4d.png)
![campaigns_notfound](https://cloud.githubusercontent.com/assets/17417959/13504870/a7aa3b9e-e133-11e5-9e63-6f00fb6bd3b2.png)

**Comment by jordan-wright** [comment]
Hey there! Thanks for the report!

That should work just fine.. All the campaign information should be self contained in the database. In fact, you can see it is, since the campaign screen loads fine to list the campaigns (that means your data should still be there, so don't worry :smile:)

What is the URL you hit when you view campaign details? Is there any hints in the console showing what the error might be?

When you ran the source version, did you see our SQL migrations occur? They should be right at the beginning of the output.

**Comment by CaptainBobby** [comment]
I did see the migrations however I am running it in a virtual environment at work that won't allow me to see the beginning of the output. 

These are the errors I get when I start the server:
 2016/03/03 13:27:03 campaign.go:148: record not found
 2016/03/03 13:27:03 campaign.go:148: record not found
 2016/03/03 13:27:03 campaign.go:148: record not found

The link I am presented with when attempting to access:
http://192.168.10.29:3333/campaigns/6
http://192.168.10.29:3333/campaigns/12
http://192.168.10.29:3333/campaigns/14

No other hints in the console other than the above. When clicking on a link to view the campaign details I get the same errors as above. It's really fine as at this point I am only in a testing phase. In the future however I will want to make sure this is working so I don't lose data.

**Comment by CaptainBobby** [comment]
I started over, created a test campaign in the binary install. Then copied the db over to the new source install and renamed it to gophish.db. I then started the server. This is the initial result:

![migrations](https://cloud.githubusercontent.com/assets/17417959/13511294/f95ec540-e151-11e5-9d26-fb9bec958516.png)

This is a capture of the errors:
![not-found](https://cloud.githubusercontent.com/assets/17417959/13511343/28837d84-e152-11e5-8560-8be5ecd41609.png)

Thanks!

**Comment by jordan-wright** [comment]
Interesting... I think I found the problem. Basically, we just introduced the concept of "sending profiles", which are stored smtp settings. There wasn't a way to migrate them over, so we're having to start fresh so to speak. This means that the sending profiles aren't found for existing campaigns. *Note: we never really stored them so they were, so it's not like you lost any data in this migration*.

I'm not sure why I haven't seen this replicated on my end, though. I'll get it fixed. Thanks for the heads up!
